libs/blocks/modal/decorators/langnav.js

```diff
@@ -0,0 +1,37 @@
+import { getConfig } from '../../../utils/utils.js';
+
+function handleEvent(prefix, link, config) {
+  document.cookie = `international=${prefix};path=/`;
+  sessionStorage.setItem('international', prefix);
+  fetch(link.href, { method: 'HEAD' }).then((resp) => {
+    if (!resp.ok) throw new Error('request failed');
+    window.location.assign(link.href);
+  }).catch(() => {
+    const prefixUrl = prefix ? `/${prefix}` : '';
+    window.location.assign(`${prefixUrl}${config.contentRoot || ''}/`);
+  });
+}
+
+function decorateLink(link, config, path) {
+  let pathname = link.getAttribute('href');
+  try { pathname = new URL(pathname).pathname; } catch (e) { /* href does not contain domain */ }
```

Comment:
It looks like all the links will usually be relative, so wrap this in an `if (pathname.startsWith('http'))` so that we're not always going through the catch.

Reply:
great idea, will do :)